src/osp/Active/SysVehicle.h

```diff
@@ -111,17 +112,77 @@ class SysVehicle : public IDynamicSystem, public IActivator
             universe::Satellite areaSat, universe::Satellite tgtSat,
             ActiveEnt tgtEnt);
 
-    /**
-     * Create a Physical Part from a PrototypePart and put it in the world
-     * @param part the part to instantiate
-     * @param rootParent Entity to put part into
-     * @return Pointer to object created
-     */
-    ActiveEnt part_instantiate(PrototypePart& part, ActiveEnt rootParent);
 
     // Handle deleted parts and separations
     void update_vehicle_modification(ActiveScene& rScene);
 
+private:
+
+    // Stores the association between a PrototypeObj entity and the indices of
+    // its owned machines in the PrototypePart array
+    struct MachineDef
+    {
+        ActiveEnt m_machineOwner;
+        std::vector<unsigned> const& m_machineIndices;
```

Comment:
are you sure you want to have a member variable that is a const-reference to a vector? that's weird.

Reply:
It is weird. The thing is, the objects of this type only live in a local function scope. In `activate_vehicle()`, there's a `PrototypePart` which owns the `PrototypeObjects` that store the machineIndices arrays. These `MachineDef` objects are created by `part_instantiate()` and filled with the machineIndices arrays, and consumed by `part_machine_instantiate()` immediately afterwards. I'm open to suggestions though.

Reply:
If its only used in one place just declare the class inside the function in question.

Otherwise put a comment in the class explaining the weird.

Reply:
It's only used in one place, except that two member functions need to return and accept them, so comment it is :P